src/main/java/com/deloitte/ddwatch/services/ProjectRepoService.java

```diff
@@ -56,21 +59,26 @@ public QualityReport getNewQualityReport(ProjectRepo projectRepo) {
 
     @Transactional
     public void updateProject(ProjectRepo projectRepo, QualityReport qualityReport) {
-        Project project = projectService.findProject(projectRepo.getProject().getId()).orElseThrow(
-                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Project not found"));
-
-        project.setQualityStatus(projectRepo.getIsDefault() ? qualityReport.getQualityStatus() : project.getQualityStatus());
+        Project project = projectRepo.getProject();
+
+        Status status;
+        LocalDateTime lastQualityReportDate;
+        if (projectRepo.getIsDefault()) {
+            status = qualityReport.getQualityStatus();
```

Comment:
You could just use the setters here, without the extra variable. The else is not necessary. If we're not dealing with a default repo, we don't need to change the project. So it has the same lastQualityReportDate and qualityStatus.